pkg/operator/awspodidentity/controller.go

```diff
@@ -0,0 +1,287 @@
+	r.logger.Debugf("reconciling after watch event %#v", request)
+	err := r.ReconcileResources()
+	if err != nil {
+		r.logger.Errorf("reconciliation failed, retrying in %s", retryInterval.String())
+		return reconcile.Result{RequeueAfter: retryInterval}, err
+	}
+	return reconcile.Result{}, nil
+}
+
+func (r *staticResourceReconciler) ReconcileResources() error {
+	applyResults := resourceapply.ApplyDirectly(
+		(&resourceapply.ClientHolder{}).WithKubernetes(r.clientset),
+		r.eventRecorder,
+		v410_00_assets.Asset,
+		staticFiles...,
+	)
+	for _, result := range applyResults {
+		if result.Error != nil {
+			r.logger.WithError(result.Error).Errorf("error reconciling %s", result.Type)
+			return result.Error
+		}
+		if !result.Changed {
+			continue
+		}
+		r.logger.Infof("%s reconciled successfully", result.Type)
+	}
+
+	// "v4.1.0/aws-pod-identity-webhook/deployment.yaml"
+	requestedDeployment := resourceread.ReadDeploymentV1OrDie(v410_00_assets.MustAsset("v4.1.0/aws-pod-identity-webhook/deployment.yaml"))
+	requestedDeployment.Spec.Template.Spec.Containers[0].Image = os.Getenv("AWS_POD_IDENTITY_WEBHOOK_IMAGE")
+	resultDeployment, modified, err := resourceapply.ApplyDeployment(r.clientset.AppsV1(), r.eventRecorder, requestedDeployment, r.deploymentGeneration, false)
+	r.deploymentGeneration = resultDeployment.Generation
+	if err != nil {
+		r.logger.WithError(err).Error("error applying Deployment")
+		return err
+	}
+	if modified {
+		r.logger.Infof("Deployment reconciled successfully")
+	}
+
+	// "v4.1.0/aws-pod-identity-webhook/mutatingwebhook.yaml"
+	requestedMutatingWebhookConfiguration := ReadMutatingWebhookConfigurationV1Beta1OrDie(v410_00_assets.MustAsset("v4.1.0/aws-pod-identity-webhook/mutatingwebhook.yaml"))
+	_, modified, err = ApplyMutatingWebhookConfiguration(r.clientset.AdmissionregistrationV1beta1(), r.eventRecorder, requestedMutatingWebhookConfiguration)
+	if err != nil {
+		r.logger.WithError(err).Error("error applying MutatingWebhookConfiguration")
+		return err
+	}
+	if modified {
+		r.logger.Infof("MutatingWebhookConfiguration reconciled successfully")
+	}
+	return nil
+}
+
+// TODO: add MutatingWebhookConfiguration helpers to library-go/operator/resource
+
+func ReadMutatingWebhookConfigurationV1Beta1OrDie(objBytes []byte) *admissionregistrationv1beta1.MutatingWebhookConfiguration {
+	requiredObj, err := runtime.Decode(defaultCodecs.UniversalDecoder(admissionregistrationv1beta1.SchemeGroupVersion), objBytes)
+	if err != nil {
+		panic(err)
+	}
+	return requiredObj.(*admissionregistrationv1beta1.MutatingWebhookConfiguration)
+}
+
+// ApplyMutatingWebhookConfiguration merges objectmeta, does not worry about anything else
+func ApplyMutatingWebhookConfiguration(client admissionregistrationclientv1beta1.MutatingWebhookConfigurationsGetter, recorder events.Recorder, required *admissionregistrationv1beta1.MutatingWebhookConfiguration) (*admissionregistrationv1beta1.MutatingWebhookConfiguration, bool, error) {
```

Comment:
As written, modification of the webhook config spec (e.g the url) by other actors will not prompt action by the operator. Is this intentional?

Reply:
yes, this is true.  i currently just ensures that the resource exists basically.

since an outside controller injects the service-ca, we really can't use `Generation` as Deployments do to detect changes and external changes are expected.

I would like to check more here.  I'll add a TODO.

Reply:
Actually, I might just add this.  Seems like if you iterate through `wehooks` and clear out `configConfig.caBundle`, we could just do a `DeepEquals` on it.

Reply:
Please review the documentation for Apply{DaemonSet,Deployment}:

https://github.com/openshift/library-go/blob/master/pkg/operator/resource/resourceapply/apps.go#L21

I think it makes sense to set the input spec hash as an annotation to enable detection of changes across operator releases. 

Regarding the use of generation to detect changes made by other actors, you're right that it can't be used in the same manner as other apply methods due to the CA field being managed by service ca. Since this behavior depends on the service ca injection annotation, it should be possible to detect when to retain the CA field of the existing resource to avoid racing. At worst this implies a second operator-initiated update when the CA is changed by service ca (to pick up the generation). I think that's a small price to pay to ensure that the operator can ensure the intended spec.

cc: @deads2k